inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp

```diff
@@ -3,455 +3,266 @@
 
-    SizeVector order;
-    SizeVector offsets(numOfDim, 0lu);
-    size_t offset = (std::numeric_limits<size_t>::max)();
-    for (size_t i = 0; i < numOfDim; i++) {
-        order.push_back(i);
+    // make primitive descriptor factory function for different configurations
+    bool dynBatchSupport = true;
+    if (axis < 1) {
+        dynBatchSupport = false;
     }
+    auto makePdInfo = [dynBatchSupport](TensorDescFactory getTensorDesc, const Precision& precision,
+                                        const MKLDNNDims& srcDims, const std::vector<MKLDNNDims>& outDims) -> PrimitiveDescInfo {
+        InferenceEngine::LayerConfig config;
+
+        config.dynBatchSupport = dynBatchSupport;
+        config.inConfs.resize(1);
+        config.inConfs[0].inPlace = -1;
+        config.inConfs[0].constant = false;
+        config.inConfs[0].desc = getTensorDesc(precision, srcDims.ToSizeVector());
+        config.outConfs.resize(outDims.size());
+
+        std::vector<memory::format> outFormats;
+
+        for (size_t i = 0; i < outDims.size(); i++) {
+            auto o_Dims = outDims[i];
 
-    SizeVector strides(numOfDim);
-    strides[numOfDim - 1] = 1;
-    for (size_t i = 2; i <= numOfDim; i++) {
-        if (numOfDim - i < axis) {
-            strides[numOfDim - i] = (std::numeric_limits<size_t>::max)();
-        } else {
-            strides[numOfDim - i] = strides[numOfDim - i + 1] * srcDims[numOfDim - i + 1];
+            config.outConfs[i].inPlace = -1;
+            config.outConfs[i].constant = false;
+            config.outConfs[i].desc = getTensorDesc(precision, o_Dims.ToSizeVector());
+            outFormats.push_back(MKLDNNMemoryDesc(config.outConfs[i].desc).getFormat());
+        }
+        return {config, impl_desc_type::ref, outFormats};
+    };
+
+    //Set plain format
+    supportedPrimitiveDescriptors.push_back(makePdInfo(&makePlainTensorDesc, inpPrecision, srcDims, outDims));
+
+    //Set per channel format
+    supportedPrimitiveDescriptors.push_back(makePdInfo(&makePerChannelTensorDesc, inpPrecision, srcDims, outDims));
+
+    //Support channel blocked format
+    std::vector<size_t> blockedPdIndexes;
+    if (srcDims.ndims() > channelsPos) {
+        for (size_t sizeS : {8lu, 16lu}) {
+            SizeVector blkDims = srcDims.ToSizeVector();
+            if (blkDims[channelsPos] % sizeS)
+                continue;
+
+            bool blocked = true;
+            for (size_t i = 0; i < outDims.size(); i++) {
+                if (outDims[i].ToSizeVector()[channelsPos] % sizeS) {
+                    blocked = false;
+                    break;
+                }
+            }
+            if (blocked) {
+                using std::placeholders::_1;
+                using std::placeholders::_2;
+                supportedPrimitiveDescriptors.push_back(makePdInfo(std::bind(&makeChannelBlockedTensorDesc, _1, _2, sizeS), inpPrecision, srcDims, outDims));
```

Comment:
If we want to move this stuff to the common scope, it may be better to use the factory method pattern. In that case the block size could be handled by one of derived from the abstract creator classes. If so, where we should move such creators?

Reply:
Discussed offline. Dedicated ticket #45051 has been created.